tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+
+# Get our discovery doc and build our service
+r = requests.get('https://cloudsupport.googleapis.com/$discovery/rest?key={}&labels=V2_TRUSTED_TESTER&version=v2alpha'.format(API_KEY))
+support_service = build_from_document(r.json())
+
+cases_file = 'support_cases.json'
+tracked_cases_file = 'tracked_cases'
+
+if os.path.exists(tracked_cases_file):
+    with open(tracked_cases_file) as tcf:
+        tracked_cases = list(eval(tcf.read()))
+else:
+    tracked_cases = []
+
+    
+class Support_Case:
+    """
+    A class used to represent a Google Cloud Support Case.
+
+    Attributes
+    ----------
+    case_number : str
+        a unique string of numbers that is the id for the case
+    resource_name : str
+        a unique string including the org or project id and the case id examples:
+        organizations/12345/cases/67890
+        projects/12345/cases/67890
+    case_title : str
+        the title the user gave the case when they created it
+    description : str
+        the user's description of the case as provided in the support ticket
+    escalated : bool
+        whether or not a case has been escalated. This field doesn't exist in 
+        the response until after a case has been escalated. True means the case
+        is escalated
+    case_creator : str
+        name of the user that opened the support case
+    create_time : str
+        timestamp of when the case was created
+    update_time : str
+        timestamp of the last update made to the case
+    priority : str
+        the current priority of the case, represented as S0, S1, S2, S3, or S4
+    state : str
+        the status of the support ticket. Can be NEW, IN_PROGRESS_GOOGLE_SUPPORT,
+        ACTION_REQUIRED, SOLUTION_PROVIDED, or CLOSED
+    comment_list : list
+        all public comments made on the case as strings. Comments are sorted with
+        newest comments at the top
+    """
+    
+    def __init__(self,caseobj):
+        """
+        Parameters
+        ----------
+        caseobj : json
+            json for an individual case
+        """
+        self.case_number = re.search('(?:cases/)([0-9]+)', caseobj['name'])[1]
+        self.resource_name = caseobj['name']
+        self.case_title = caseobj['displayName']
+        self.description = caseobj['description']
+        if 'escalated' in caseobj:
+            self.escalated = caseobj['escalated']
+        else:
+            self.escalated = False
+        self.case_creator = caseobj['creator']['displayName']
+        self.create_time = str(datetime.fromisoformat(
+            caseobj['createTime'].replace('Z', '+00:00')))
+        self.update_time = str(datetime.fromisoformat(
+            caseobj['updateTime'].replace('Z', '+00:00')))
+        self.priority = caseobj['severity'].replace('S', 'P')
+        self.state = caseobj['state']
+        req = support_service.cases().comments().list(parent=self.resource_name)
+        self.comment_list = req.execute().get('comments',[])
+
```

Comment:
Can you add max_retries=3 to execute method toimprove reliability?

Reply:
Added!